To: vim_dev@googlegroups.com
Subject: Patch 7.3.282
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.282
Problem:    When using input() and :echo in a loop the displayed text is
	    incorrect. (Benjamin Fritz)
Solution:   Only restore the cursor position when there is a command line.
	    (Ben Schmidt)
Files:	    src/ex_getln.c


*** ../vim-7.3.281/src/ex_getln.c	2011-07-27 17:58:42.000000000 +0200
--- src/ex_getln.c	2011-08-17 15:57:39.000000000 +0200
***************
*** 1969,1976 ****
  # endif
      s = getcmdline(firstc, 1L, 0);
      restore_cmdline(&save_ccline);
!     /* Restore msg_col, the prompt from input() may have changed it. */
!     msg_col = msg_col_save;
  
      return s;
  }
--- 1969,1980 ----
  # endif
      s = getcmdline(firstc, 1L, 0);
      restore_cmdline(&save_ccline);
!     /* Restore msg_col, the prompt from input() may have changed it.
!      * But only if called recursively and the commandline is therefore being
!      * restored to an old one; if not, the input() prompt stays on the screen,
!      * so we need its modified msg_col left intact. */
!     if (ccline.cmdbuff != NULL)
! 	msg_col = msg_col_save;
  
      return s;
  }
*** ../vim-7.3.281/src/version.c	2011-08-17 15:23:16.000000000 +0200
--- src/version.c	2011-08-17 16:24:04.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     282,
  /**/

-- 
Why don't cannibals eat clowns?
Because they taste funny.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
